keps/sig-network/20191205-kube-proxy-IP-node-binding.md

```diff
@@ -0,0 +1,75 @@
+2. Some LBs are sending packet with the destination IP equal to the LB IP, so the IP needs to be routable directly to the right service (otherwise looping back and forth)
+3. (TBD)
+
+This enhancement proposes a way to make this features configurable, giving the cloud provider, or the user, a way to disable this behaviour of kube proxy.
+The best way would be to set a defaut behaviour at the cloud controller level, that can be overwritten bu the user at a service level.
+
+## Motivation
+
+There are numerous problems with the current behaviour:
+
+1. Some cloud provider (Scaleway, ...) are sometimes using the LB's external IP as source IP when sending packets to the cluster. This is a probem in the ipvs mode of kube proxy since the IP is bouded to an interface and healtchecks from the LB are never coming back.
+2. Some cloud provider (DigitalOcean, Scaleway, ...) are also having features at the LB level (TLS termination, PROXY protocol, ...) and bypassing the LB means missing these features when the packet arrives to the service (leading to protocols errors).
+
+So giving an options to these cloud to disable the actual behaviour would be very valuable.
+
+Currently there is some hacky workaround that set the `Hostname` on the `Service` in order to bypass kube proxy binding (AWS and DigitalOcean for instance), but as stated, it's a bit hacky. This KEP should also give a solution to this hack.
+
+### Goals
+
+* Expose an option (ideally for at least the cloud controller manager) to disable the actual behaviour of kube proxy by default for all services.
+* Expose an annotation or a label on the `Service` to give the user the ability to enable/disable this feature on specific service.
+
+### Non-Goals
+
+* Deprecate the actual behaviour of kube proxy
+* Change the default actual behaviour of kube proxy
+
+## Proposal
+
+The idea is to add a new well known annotation or label `service.kubernetes.io/disable-lb-bypass` (not the definitive name). This label/annotation will be able to controle the `kube-proxy` behaviour for a specific service and the rules associated (especially disable the LB bypassing, leading all request going through the LB's IP to effectively go though the LB).
+
+This change on its own is not enough since it will force the user to add this label/annotation on service that requires disable LB bypass. 
+
+The second idea is to add a new method to the Cloud Controller Manager LoadBalancer interface (`DisableLBBypass(ctx context.Context) bool` for instance) that returns the default behaviour for the given cloud provider. Then the cloud controller will call this method when the `Service` becomes a `LoadBalancer` service, and setting the previous label/annotation to the desired value is the user did not already specify it.
+Finally the `kube-proxy` will be able to take the decision to bypass the LB or not base on the value of this label/annotation.
```

Comment:
Finally the kube-proxy will be able to ~take~ `make` the decision to bypass the LB or not, ~base~ `based` on the value of this label/annotation.